src/estimagic/optimization/tranquilo/sample_points.py

```diff
@@ -110,47 +117,207 @@ def _naive_sampler(
             x vector at which the criterion function has already been evaluated, that
             satisfies lower_bounds <= existing_xs <= upper_bounds.
         rng (numpy.random.Generator): Random number generator.
-        bounds (Bounds or None): NamedTuple
+        bounds (Bounds or None): NamedTuple.
 
     """
-    n_points = _get_effective_n_points(target_size, existing_xs)
+    n_points = _get_effective_n_points(target_size, existing_xs=existing_xs)
     n_params = len(trustregion.center)
-    region_bounds = _get_effective_bounds(trustregion, bounds)
+    effective_bounds = _get_effective_bounds(trustregion, bounds=bounds)
 
     points = rng.uniform(
-        low=region_bounds.lower,
-        high=region_bounds.upper,
+        low=effective_bounds.lower,
+        high=effective_bounds.upper,
         size=(n_points, n_params),
     )
+    return points
+
+
+def _hull_sampler(
+    trustregion,
+    target_size,
+    rng,
+    ord,  # noqa: A002
+    existing_xs=None,
+    bounds=None,
+):
+    """Random generation of trustregion points on the hull of general sphere / cube.
+
+    Points are sampled randomly on a hull (of a sphere for ord=2 and of a cube for
+    ord=np.inf). These points are then mapped into the feasible region, which is defined
+    by the intersection of the trustregion and the bounds.
+
+    Args:
+        trustregion (TrustRegion): NamedTuple with attributes center and radius.
+        target_size (int): Target number of points in the combined sample of existing_xs
+            and newly sampled points. The sampler does not have to guarantee that this
+            number will actually be reached.
+        rng (numpy.random.Generator): Random number generator.
+        ord (int): Type of norm to use when scaling the sampled points. For 2 it will
+            result in sphere sampling, for np.inf in cube sampling.
+        existing_xs (np.ndarray or None): 2d numpy array in which each row is an
+            x vector at which the criterion function has already been evaluated, that
+            satisfies lower_bounds <= existing_xs <= upper_bounds.
+        bounds (Bounds or None): NamedTuple.
 
+    """
+    n_points = _get_effective_n_points(target_size, existing_xs=existing_xs)
+    n_params = len(trustregion.center)
+    effective_bounds = _get_effective_bounds(trustregion, bounds=bounds)
+
+    points = rng.normal(size=(n_points, n_params))
```

Comment:
I think a uniform might be better for box sampling. Would make it flexible for now so we can try it out in a benchmark.